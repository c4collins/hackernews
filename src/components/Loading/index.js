// Core Modules
import React from 'react';

// Constants
import { FontAwesome } from '../../constants';

// Misc Files
import './index.css';

const Loading = () =>
    <FontAwesome
        name='spinner'
        size='2x'
        spin
        style={{ 
            textShadow: '0 1px 0 rgba(0, 0, 0, 0.1)',
            color: 'blue',
        }}
    />

const withLoading = (Component) =>
    ({isLoading, ...rest}) =>
        isLoading 
        ? <Loading/> 
        : <Component {...rest} />

export default Loading;
export { withLoading };