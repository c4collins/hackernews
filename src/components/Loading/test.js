import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Loading from './index';

describe('Loading', () => {

  it('renders', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Loading />, div);
  });

  test('snapshots', () => {
    const component = renderer.create(
      <Loading />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});