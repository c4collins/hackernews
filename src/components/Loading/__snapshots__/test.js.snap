// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading snapshots 1`] = `
<span
  aria-hidden={true}
  className="fa fa-spinner fa-2x fa-spin"
  style={
    Object {
      "color": "blue",
      "textShadow": "0 1px 0 rgba(0, 0, 0, 0.1)",
    }
  }
/>
`;
